share/spice/twitch/featured/twitch_featured.js

```diff
@@ -0,0 +1,44 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_twitch_featured = function(api_result){
+
+        if (!api_result || !api_result.featured || api_result.featured.length === 0 ||api_result.error) {
+            return Spice.failed('twitch_featured');
+        }
+        
+        Spice.add({
+            minItemsForModeSwitch: 3,
+            id: "twitch_featured",
+            name: "Twitch Featured Streams",
+            data: api_result.featured,
+            meta: {
+                sourceName: "twitch.tv",
```

Comment:
Nit: it might look better if it was just "Twitch" and it's more consistent with our other IAs (also, that's what they have in the title):

![screen shot 2015-04-07 at 11 15 22 pm](https://cloud.githubusercontent.com/assets/81969/7038195/027b4f8e-dd7c-11e4-81b3-ccca8c7ce55a.png)

![screen shot 2015-04-07 at 11 14 52 pm](https://cloud.githubusercontent.com/assets/81969/7038200/0ab0a730-dd7c-11e4-9b5a-642f12deada6.png)
![screen shot 2015-04-07 at 11 14 59 pm](https://cloud.githubusercontent.com/assets/81969/7038199/0ab02a62-dd7c-11e4-9af2-46047de6309a.png)